tools/cmake/component.cmake

```diff
@@ -379,7 +379,11 @@ function(idf_component_get_property var component property)
     if(__GENERATOR_EXPRESSION)
         set(val "$<TARGET_PROPERTY:${component_target},${property}>")
     else()
-        __component_get_property(val ${component_target} ${property})
+        if("${component_target}" STREQUAL "")
```

Comment:
I think we have to do this before the `__GENERATOR_EXPRESSION` check, so that the correct error is also given for the `__GENERATOR_EXPRESSION` case.